src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -0,0 +1,144 @@
+                let parts = fullModuleName.Split('.')
+                if parts.Length = 1 then ("global", parts.[0])
+                else (String.Join(".", parts, 0, parts.Length - 1), parts.[parts.Length - 1])
+            let generateGetObject =
+                match _targetFramework with
+                | "netstandard1.0"
+                | "netstandard1.1"
+                | "netstandard1.2"
+                | "netstandard1.3"
+                | "netstandard1.4"
+                | "netstandard1.5"
+                | "netstandard1.6" -> false // these targets don't support the `ResourceManager.GetObject()` method
+                | _ -> true // other supported runtimes, do
+            let sb = StringBuilder().AppendLine(String.Format(boilerplate, namespaceName, moduleName, justFileName))
+            if generateGetObject then sb.AppendLine(boilerplateGetObject) |> ignore
+            let sourcePath = Path.Combine(_outputPath, justFileName + ".fs")
+            printMessage <| sprintf "Generating: %s" sourcePath
+            let body =
+                let xname = XName.op_Implicit
+                XDocument.Load(resx).Descendants(xname "data")
+                |> Seq.fold (fun (sb:StringBuilder) (node:XElement) ->
+                    let name =
+                        match node.Attribute(xname "name") with
+                        | null -> failwith "Missing resource name"
+                        | attr -> attr.Value
+                    let docComment =
+                        match node.Elements(xname "value").FirstOrDefault() with
+                        | null -> failwith <| sprintf "Missing resource value for '%s'" name
+                        | element -> element.Value.Trim()
+                    let identifier = if Char.IsLetter(name.[0]) || name.[0] = '_' then name else "_" + name
+                    let commentBody =
+                        XElement(xname "summary", docComment).ToString().Split([|"\r\n"; "\r"; "\n"|], StringSplitOptions.None)
+                        |> Array.fold (fun (sb:StringBuilder) line -> sb.AppendLine("    /// " + line)) (StringBuilder())
+                    // add the resource
+                    let accessorBody =
+                        match (generateLegacy, generateLiteral) with
+                        | (true, true) -> sprintf "    [<Literal>]\n    let %s = \"%s\"" identifier name
+                        | (true, false) -> sprintf "    let %s = \"%s\"" identifier name // the [<Literal>] attribute can't be used for FSharp.Core
+                        | (false, _) ->
+                            let isStringResource = match node.Attribute(xname "type") with
+                                                   | null -> true
+                                                   | _ -> false
+                            match (isStringResource, generateGetObject) with
+                            | (true, _) -> sprintf "    let %s() = GetString(\"%s\")" identifier name
```

Comment:
Just a thought, but could we replace `@"\n"` with `"\n"` at this point? Also it would be really neat if we could detect `{0},{1}` etc. in the string and then make a function take them as parameters. Such as:

```f#
let MyString (parameter1 : string) = String.Format(GetString("MyString"), parameter1)
```

It means string formatting identifiers in the resx will fail the build if they aren't expanded.

Reply:
Regarding the `@"\n" -> "\n"` replacement, I'd rather not do it here because I would like this to be as close to the C#/VB implementation as possible and that only outputs `ResourceManager.GetString()`.  Also, it seems wrong to do any resource manipulation without the call site explicitly knowing about it, especially since each fetch of a string resource now contains an extra function call and potentially more string allocations.

Reply:
Some of the SR modules that you removed did this already - although am I right in saying that this code is for *all* FSharp projects, not just VFT projects?

Reply:
As for detecting `{0},{1}`, that's a surprisingly hard problem.  In the past a Roslyn team member tried to write a C#/VB analyzer to detect mismatches between the value to `String.Format()` and the replacement values.  That ended up being abandoned because to make it correct the entire string parsing had to be re-implemented and it gets very nasty very fast.